docs/flink-getting-started.md

```diff
@@ -634,6 +634,26 @@ FlinkSink.forRowData(input)
 env.execute("Test Iceberg DataStream");
 ```
 
+### Streaming upsert write
+
+Iceberg supports `UPSERT` based on the primary key when writing streaming data into v2 table format. To enable it, create an `UPSERT` table or configuration it when building a writer.
+
+For an `UPSERT` table, all writing will be in `UPSERT` mode unless you overwrite it when building the writer:
+
+```
+CREATE TABLE `hive_catalog`.`default`.`sample` (
+  `id`  INT UNIQUE COMMENT 'unique id',
+  `data` STRING NOT NULL,
+ PRIMARY KEY(`id`) NOT ENFORCED
+) with ('format-version'='2', 'write.upsert.enabled'='true');
+```
+
+Enabling `UPSERT` using `upsert-enabled` in the [write options](#Write options) provides more flexibility than when creating a table. Note that you still need to use v2 table format and specify the primary key when creating the table.
+
+{{< hint info >}}
+OVERWRITE mode shouldn't be enable when configuring to use UPSERT data stream. In UPSERT mode, if the table is partitioned, the partition fields should be a part of equality fields.
```

Comment:
nit: 

`OVERWRITE mode shouldn't be enable when configuring to use UPSERT data stream` 
-> 
`OVERWRITE and UPSERT can't be set together`

Reply:
nit: `should be a part of equality fields` -> `should be included in equality fields`